### PR TITLE
Added HttpClientBuilderWithoutPoolSize

### PR DESCRIPTION
**Pull request by stokpop**
This is a fix for issue: https://github.com/jborgers/PMD-jPinpoint-rules/issues/91

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ You don't have to wait for that, you can already use these as custom rules right
 The jpinpoint rules can be run from the command-line using the PMD tool, from your favorite development
 environment with a PMD-plugin, or in SonarQube after packaging them as Sonar plugin.
 
+Documentation of the rules are here:
+* [JavaCodePerformance](docs/JavaCodePerformance.md)
+* [JavaDataAccessPerformance](docs/JavaDataAccessPerformance.md)
+
```
Comment (jborgers) [inline README.md:18]: Please add the third doc page, as shown under Wiki

File: src/main/resources/category/java/common.xml [hunk 2]
```diff
@@ -823,6 +823,40 @@ ancestor::MethodDeclaration//PrimarySuffix/@Image="disableConnectionState")]
...
+(: method param http client builder without setMaxConnPerRoute :)
+//MethodDeclaration//FormalParameter//ClassOrInterfaceType[
+(@Image="HttpClientBuilder" or @Image="HttpAsyncClientBuilder" or @Image="HttpClients") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnTotal")]
+			]]>
+                </value>
+            </property>
+        </properties>
+    </rule>
+
```
Comment (jborgers) [inline src/main/resources/category/java/common.xml:859]: Please use pmd-java:typeIs() for an exact match of the type.

Comment (jborgers) [inline src/main/resources/category/java/common.xml:859]: Can you combine the first two and last two blocks and use 'or' to be more concise?

Comment (jborgers) [inline src/main/resources/category/java/common.xml:859]: Please add example code for good and bad situation

**Review by jborgers — CHANGES_REQUESTED** [review]
see inline remarks

File: src/main/resources/category/java/common.xml (hunk 2 above)
Comment (stokpop) [reply src/main/resources/category/java/common.xml:859]: We tried, but then only one violation shows when actually two setters are missing. Seems more concise to have one violation per setter.

File: README.md (hunk 1 above)
Comment (stokpop) [reply README.md:21]: added

File: src/main/resources/category/java/common.xml (hunk 2 above)
Comment (stokpop) [reply src/main/resources/category/java/common.xml:968]: added

Comment (stokpop) [reply src/main/resources/category/java/common.xml:968]: done

**Comment by stokpop** [comment]
There still seems to be a lot of replication/repetition in this new rule definitions, if you see any room for improvement, let me know.

File: src/main/resources/category/java/common.xml [hunk 3]
```diff
@@ -823,6 +823,40 @@ ancestor::MethodDeclaration//PrimarySuffix/@Image="disableConnectionState")]
...
+//MethodDeclaration//PrimaryExpression/PrimaryPrefix/Name[
+(@Image="HttpClientBuilder.create" or @Image="HttpAsyncClientBuilder.create" or @Image="HttpClients.custom") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnTotal")]
+,
+(: method param http client builder without setMaxConnPerRoute :)
+//MethodDeclaration//FormalParameter//ClassOrInterfaceType[
+(@Image="HttpClientBuilder" or @Image="HttpAsyncClientBuilder" or @Image="HttpClients") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnPerRoute")]
+,
+(: method param http client builder without setMaxConnPerRoute :)
```
Comment (jborgers) [inline src/main/resources/category/java/common.xml:918]: docu incorrect

**Review by jborgers — CHANGES_REQUESTED** [review]
fix typo in xpath doc, see my remark

**Review by jborgers — APPROVED** [review]

**Pull request merged**